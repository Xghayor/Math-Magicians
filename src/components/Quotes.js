import React, { useState, useEffect } from 'react';

function Quotes() {
  const [data, setData] = useState([]);
  const [hasError, setHasError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    setIsLoading(true);
    const fetchData = async () => {
      try {
        const headers = {
          'X-Api-Key': '3DPT6zJuhp6ivWqildtvnA==aOy3cSTtq3BUgEhT',
        };

        const res = await fetch('https://api.api-ninjas.com/v1/quotes?category=happiness', {
          method: 'GET',
          headers,
        });

        if (!res.ok) {
          throw new Error('Network response was not ok');
        }

        const json = await res.json();
        setData(json);
        setIsLoading(false);
      } catch (error) {
        setHasError(true);
        setIsLoading(false);
      }
    };

    fetchData();
  }, []);

  if (hasError) return <div>Something went wrong!</div>;

  if (isLoading) {
    return (
      <div className="bubble-spinner">
        <div className="bubble" />
        <div className="bubble" />
        <div className="bubble" />
      </div>
    );
  }

  return (
    <ul>
      {data.map((item) => (
        <li key={`${item.id}`}>
          {item.quote}
          <br />
          <strong>{item.author}</strong>
        </li>
      ))}

    </ul>
  );
}

export default Quotes;
