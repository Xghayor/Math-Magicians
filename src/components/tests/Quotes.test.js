import React from 'react';
import { render } from '@testing-library/react';
import Quotes from '../Quotes';

test('Quotes component matches snapshot', () => {
  const { asFragment } = render(<Quotes />);
  expect(asFragment()).toMatchSnapshot();
});
