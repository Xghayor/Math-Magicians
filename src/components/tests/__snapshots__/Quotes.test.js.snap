// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bubble-spinner"
  >
    <div
      class="bubble"
    />
    <div
      class="bubble"
    />
    <div
      class="bubble"
    />
  </div>
</DocumentFragment>
`;
